logstash-core/lib/logstash/agent.rb

```diff
@@ -109,6 +116,12 @@ def node_uuid
     @node_uuid ||= SecureRandom.uuid
   end
 
+  def running_pipelines?
+    @upgrade_mutex.synchronize do
+      @pipelines.select {|pipeline_id, _| running_pipeline?(pipeline_id) }.any?
+    end
+  end
+
   private
   def start_webserver
```

Comment:
Check the error we are getting from manticore, maybe related to starting webserver